agent/release/billi-release.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+function verify_if_release_image_exists() {
+    local release_image=$1
+    if [ ! "$(podman manifest inspect ${release_image})" ]; then
+        echo "Cannot download image ${release_image}"
+    fi
+}
+
+function clone_agent_repo() {
+    
+    local commit=$1
+
+    if [[ ! -d "installer" ]]; then 
+        git clone https://github.com/openshift/installer.git
+    fi
+    
+    if [[ ! -z "${commit}" ]]; then
+        echo "Building from commit $commit"
+    fi
+
+    cd installer
+    git checkout $commit
+    cd ..
+}
+
+function build_agent_installer() {
+    cd installer
+    hack/build.sh
+    cd ..
+}
+
+function patch_openshift_install_release_version() {
+    local version=$1
+
+    local res=$(grep -oba ._RELEASE_VERSION_LOCATION_.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX installer/bin/openshift-install)
+    local location=${res%%:*}
+
+    # If the release marker was found then it means that the version is missing
+    if [[ ! -z ${location} ]]; then
+        echo "Patching openshift-install with version ${version}"
+        printf "${version}\0" | dd of=installer/bin/openshift-install bs=1 seek=${location} conv=notrunc &> /dev/null 
+    else
+        echo "Version already patched"
+    fi
+}
+
+function patch_openshift_install_release_image() {
+    local image=$1
+
+    local res=$(grep -oba ._RELEASE_IMAGE_LOCATION_.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX installer/bin/openshift-install)
+    local location=${res%%:*}
+
+    # If the release marker was found then it means that the image is missing
+    if [[ ! -z ${location} ]]; then
+        echo "Patching openshift-install with image ${image}"
+        printf "${image}\0" | dd of=installer/bin/openshift-install bs=1 seek=${location} conv=notrunc &> /dev/null 
+    else
+        echo "Image already patched"
+    fi
+}
+
+function complete_release() {
+    echo 
+    echo "BILLI installer release completed"
+    echo 
+
+    installer/bin/openshift-install version
+}
+
+if [ "$#" -ne 3 ]; then
+    echo "Usage: billi-release.sh <commit> <release version> <release image>"
+    echo "[Arguments]"
+    echo "      <commit>            The git commit to be used for building the installer"
+    echo "      <release version>   The desired OpenShift release version"
+    echo "      <release image>     The desired OpenShift release image location"
+    echo
+    echo "Important note: the specified <release version> must match the one pointed by <release image>"
+    echo
+    echo "example: ./billi-release.sh 38e719a 4.11.0 quay.io/openshift-release-dev/ocp-release@sha256:300bce8246cf880e792e106607925de0a404484637627edf5f517375517d54a4"
+
+    exit 1
+fi
+
+commit=$1
+release_version=$2
```

Comment:
I think this can be simplified. The release version can be determined from the release image using:

oc adm release info -o template --template '{{.metadata.version}}' --insecure=true quay.io/openshift-release-dev/ocp-release@sha256:300bce8246cf880e792e106607925de0a404484637627edf5f517375517d54a4

Reply:
Thanks @rwsu it seems really a good idea!

Reply:
I like this but we should probably also document how to log in to the registry.